var/spack/repos/spack_repo/builtin/packages/rust/package.py

```diff
@@ -217,6 +217,11 @@ def configure(self, spec, prefix):
         # Compile tools into flag for configure.
         flags.append(f"--tools={','.join(tools)}")
 
+        # by default a `-nightly` suffix we be applied even on
+        # non-nightly stable builds
+        if self.spec.version != Version("nightly"):
+            flags.append("--release-channel=stable")
```

Comment:
```suggestion
        if not self.spec.satisfies("@=nightly"):
            flags.append("--release-channel=stable")
```

We should instead use `spec.satisfies()` instead of creating a version object.